x/peggy/abci.go

```diff
@@ -264,6 +302,25 @@ func (h *BlockHandler) valsetSlashing(ctx sdk.Context, params *types.Params) {
 	}
 }
 
+// prepBatchConfirms loads all confirmations into a hashmap indexed by validatorAddr
+// reducing the lookup time dramatically and separating out the task of looking up
+// the orchestrator for each validator
+func prepBatchConfirms(ctx sdk.Context, k keeper.Keeper, batch *types.OutgoingTxBatch) map[string]*types.MsgConfirmBatch {
+	confirms := k.GetBatchConfirmByNonceAndTokenContract(ctx, batch.BatchNonce, common.HexToAddress(batch.TokenContract))
+	// bytes are incomparable in go, so we convert the sdk.ValAddr bytes to a string (note this is NOT bech32)
+	ret := make(map[string]*types.MsgConfirmBatch)
+	for _, confirm := range confirms {
+		// TODO this presents problems for delegate key rotation see issue #344
```

Comment:
Why is this a problem? Does this now break orch address rotation??

Reply:
I assume that it's because `GetOrchestratorValidator` won't find anything if we changed the orch address. I couldn't find that issue.
It's not a new issue introduced here, but before we didn't have that comment and they did: https://github.com/Gravity-Bridge/Gravity-Bridge/commit/7cf99c4f09bcfaf93b2b0b56a0c8bd18a5fc5fed#diff-78fbe3fdd3742b985b418da2306930ee8a83fee765a4a743428a0dc61903f072L220

Reply:
Where is issue # 344? Is the issue that the claim could be for an old orch address that no longer exists?

Reply:
@alexanderbez found it https://github.com/cosmos/gravity-bridge/issues/344 I get lost with so many repos lol

Reply:
So does that mean that key rotation is just now broken or not supported in GB? Or will it be broken only when this case hits?

Reply:
They don't support it 😬 
https://github.com/Gravity-Bridge/Gravity-Bridge/blob/5507f0744ba9e66dad08cffb894782e3e5d14c84/module/x/gravity/keeper/msg_server.go#L43-L52
https://github.com/Gravity-Bridge/Gravity-Bridge/blob/5507f0744ba9e66dad08cffb894782e3e5d14c84/module/x/gravity/types/errors.go#L17

Reply:
Ok so we're going to have to undo key rotation then. Can you open an issue please?

Reply:
https://github.com/umee-network/umee/issues/364